llmfoundry/utils/config_utils.py

```diff
@@ -341,8 +341,6 @@ def make_dataclass_and_log_config(
         transforms,
     )
 
-    logged_cfg.update(unstructured_config, merge=True)
```

Comment:
I think you want to move the `logged_cfg = copy.deepcopy(unstructured_config)` line down here after the transforms right?

Reply:
I considered both options, and decided to err on the side of reproducibility. In particular, I was thinking that if there is a faulty transformation, then once the data is logged post-transformation there is no way to get back the original config and reproduce the issue. In that case it would be preferable to log the original config to be able to reproduce the same behavior .

Reply:
hmm ok. i guess i can see it both ways, but i accept.